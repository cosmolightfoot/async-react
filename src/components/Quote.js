import React from 'react';
import PropTypes from 'prop-types';

function Quote({ quote, character, image }) {
  return (
    <li>
      <img src={image} />
      <h3>{character}</h3>
      <p>{quote}</p>
    </li>
  );
}

Quote.propTypes = {
  quote: PropTypes.string.isRequired,
  character: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Quote;
