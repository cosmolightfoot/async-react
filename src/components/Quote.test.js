import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote snapshot test', () => {
  it('snapshots', () => {
    const wrapper = shallow(<Quote character="bender" quote="bite me" image="www.url.com" />);
    expect(wrapper).toMatchSnapshot();
  });
});
