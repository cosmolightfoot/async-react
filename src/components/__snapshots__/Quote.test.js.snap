// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote snapshot test snapshots 1`] = `
<li>
  <img
    src="www.url.com"
  />
  <h3>
    bender
  </h3>
  <p>
    bite me
  </p>
</li>
`;
